apps/console/src/features/roles/components/edit-role/edit-role-groups-updated.tsx

```diff
@@ -0,0 +1,547 @@
+            dispatch(
+                addAlert({
+                    description: t("console:manage.features.roles.edit.groups.notifications.fetchError.description"),
+                    level: AlertLevels.ERROR,
+                    message: t("console:manage.features.roles.edit.groups.notifications.fetchError.message")
+                })
+            );
+        }
+    }, [ groupListFetchRequestError ]);
+
+    /**
+     * Handle the restore groups.
+     * 
+     * @param remainingGroups - remaining groups
+     */
+    const handleRestoreGroups = (remainingGroups: GroupsInterface[]) => {
+        const removedGroups: GroupsInterface[] = [];
+
+        removedGroupsOptions.forEach((group: GroupsInterface) => {
+            if (!remainingGroups.find((newGroup: GroupsInterface) => newGroup.id === group.id)) {
+                removedGroups.push(group);
+            }
+        });
+
+        setSelectedGroupsOptions([
+            ...selectedGroupsOptions,
+            ...removedGroups
+        ]);
+    };
+
+    /**
+     * Get the place holder components.
+     * 
+     * @returns - place holder components
+     */
+    const getPlaceholders = () => {
+        if (groupListFetchRequestError) {
+            return (
+                <EmptyPlaceholder
+                    subtitle={ 
+                        [ t("console:manage.features.roles.edit.groups.placeholders.errorPlaceholder.subtitles.0"),
+                            t("console:manage.features.roles.edit.groups.placeholders.errorPlaceholder.subtitles.1") ] 
+                    }
+                    title={ t("console:manage.features.roles.edit.groups.placeholders.errorPlaceholder.title") }
+                    image={ getEmptyPlaceholderIllustrations().genericError }
+                    imageSize="tiny"
+                />
+            );
+        } else if (showEmptyRolesListPlaceholder) {
+            return (
+                <EmptyPlaceholder
+                    subtitle={ 
+                        [ t("console:manage.features.roles.edit.groups.placeholders.emptyPlaceholder.subtitles.0") ]
+                    }
+                    title={ t("console:manage.features.roles.edit.groups.placeholders.emptyPlaceholder.title") }
+                    image={ getEmptyPlaceholderIllustrations().emptyList }
+                    imageSize="tiny"
+                    action={ 
+                        !isReadOnly 
+                            ? (
+                                <Button onClick={ () => setShowEmptyRolesListPlaceholder(false) } >
+                                    { t("console:manage.features.roles.edit.groups.placeholders.emptyPlaceholder" + 
+                                        ".action") }
+                                </Button>
+                            )
+                            : null
+                    }
+                />
+            );
+        }
+    };
+
+    /**
+     * The following function handles the search query for the groups list.
```

Comment:
Let's make the comments consistent
```suggestion
     *Handle the search query for the groups list.
```

Reply:
Fixed with https://github.com/wso2/identity-apps/pull/4212/commits/c57c7b953b186b80c95a389a9c46805020b584fa